### PR TITLE
Fixed issue with size not being updated in TreeMap upon removal of so…

### PR DESCRIPTION
**Pull request by shannah**
…me nodes.

For issue https://github.com/konsoletyper/teavm/issues/112

### CODE REVIEW

File: teavm-classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java [hunk 1]
```diff
@@ -363,8 +363,8 @@ public boolean containsKey(Object key) {
             while (minDepth > 0) {
                 TreeNode<K, V> node = pathToMin[--minDepth];
                 node.left = right;
-                node.fix();
                 node = node.balance();
+                node.fix();
```
Comment (konsoletyper) [inline teavm-classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java]: This replacement is not necessary. balance() already fixes heights.

Comment (shannah) [inline teavm-classlib/src/main/java/org/teavm/classlib/java/util/TTreeMap.java]: If the node is already balanced, then balance() won't touch the height.

**Comment by konsoletyper** [comment]
Yes, but heigts are invalid before balance(). So you should not have removed it.

**Comment by shannah** [comment]
I see your point.  This particular change may not have been necessary.  I see your change, and I'm in the process of testing it.

**Pull request merged**